### PR TITLE
Stop PMON docker before cold and soft reboots

### PR DESCRIPTION
**Pull request by aravindmani-1**
#### What I did
Stopped PMON before cold boot
#### How I did it
Edit reboot script and stop pmon docker and services
#### How to verify it
Check docker ps output just before cold boot is being executed.  
#### New command output (if the output of a command-line utility has changed)
UT:
[cold_boot_UT.txt](https://github.com/Azure/sonic-utilities/files/6161266/cold_boot_UT.txt)

### CODE REVIEW

**Comment by aravindmani-1** [comment]
@jleveque Please review this PR.

File: scripts/reboot [hunk 1]
```diff
@@ -178,6 +190,8 @@ if [ -x ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE} ]; then
     ${DEVPATH}/${PLATFORM}/${PLATFORM_UPDATE_REBOOT_CAUSE}
 fi
 
+stop_pmon_service
```
Comment (jleveque) [inline scripts/reboot:193]: I understand you want to stop this at the last possible moment, but does it just make things simpler to add the logic to the existing `stop_sonic_services()` function?

**Review by jleveque — CHANGES_REQUESTED** [review]
Once all comments are addressed, please also add the same logic to the new soft-reboot script as part of this PR: `scripts/soft-reboot`

File: scripts/reboot (hunk 1 above)
Comment (aravindmani-1) [reply scripts/reboot:193]: If there is any platform API is introduced/invoked in between after stopping PMON, it will fail if PMON is shutdown early.

Comment (yxieca) [reply scripts/reboot:193]: I don't see a problem moving stop_sonic_service to here.

Comment (yxieca) [reply scripts/reboot:193]: @aravindthoram can we consolidate the change in stop_sonic_services and move the call to there?

Comment (aravindmani-1) [reply scripts/reboot:193]: @yxieca Will commit the new changes soon.

**Comment by aravindmani-1** [comment]
@yxieca , @jleveque  can you please review the latest changes?.

File: scripts/reboot [hunk 2]
```diff
@@ -51,16 +51,29 @@ function tag_images()
...
     if [[ x"$SUBTYPE" == x"DualToR" ]]; then
         debug "DualToR detected, stopping mux container before reboot..."
         systemctl stop mux
     fi
+
     if [[ x"$ASIC_TYPE" != x"mellanox" ]]; then
         debug "Stopping syncd process..."
         docker exec -i syncd /usr/bin/syncd_request_shutdown --cold > /dev/null
         sleep 3
+        stop_pmon_service
```
Comment (yxieca) [inline scripts/reboot:76]: I think you should put stop_pmon_service outside the if block. Why skipping it for Mellanox platforms?

**Comment by aravindmani-1** [comment]
@yxieca Addressed the comments.

**Review by yxieca — APPROVED** [review]

**Review by jleveque — APPROVED** [review]

**Pull request merged**